cf/data/data.py

```diff
@@ -12021,16 +12036,21 @@ def log(self, base=None, inplace=False, i=False):
 
         """
         d = _inplace_enabled_define_and_cleanup(self)
+        dx = d._get_dask()
 
         if base is None:
-            d.func(np.log, units=_units_1, inplace=True)
+            dx = da.log(dx)
         elif base == 10:
-            d.func(np.log10, units=_units_1, inplace=True)
+            dx = da.log10(dx)
         elif base == 2:
-            d.func(np.log2, units=_units_1, inplace=True)
+            dx = da.log2(dx)
         else:
-            d.func(np.log, units=_units_1, inplace=True)
-            d /= np.log(base)
+            dx = da.log(dx)
+            dx /= da.log(base)
+
+        d._set_dask(dx, reset_mask_hardness=False)
+
+        d._Units = _units_1  # all logarithm outputs are unitless
```

Comment:
This is nice-ifying the old code, but this might be better (signals intent and avoids explicit underscore access):

```suggestion
        d.override_units(_units_1, inplace=True)  # all logarithm outputs are unitless
```